packages/server/src/helpers/files-validator.ts

```diff
@@ -1,68 +1,101 @@
     }
-    else if (!isAddress(owner)) {
-      errors.push({ owner: 'Given address is not valid' })
+    else if (!utils.isAddress(owner)) {
+      errors.push({ owner: 'Given owner address is not valid' })
     }
     if (!file) {
       errors.push({ file: 'File content must be given' })
     }
     if (errors.length) ctx.throw(BAD_REQUEST, {errors})
   }
 
-  // delete endpoint to do later
-  // static validateForDelete: publicValidateFunction = (ctx) => {
-  //   FilesValidator.validateFileExists(ctx)
-  //   const errors: error[] = []
-  //   const { signature } = ctx.request.body
-  //   if (!signature) {
-  //     errors.push({ signature: 'File owner signature must be given' })
-  //   }
-  //   // todo: { signature: 'Given signature is not valid' }
-  //   if (errors.length) ctx.throw(BAD_REQUEST, {errors})
-  // }
+  static async validateForDelete(ctx: Context) {
+    await this.validateFileExists(ctx)
+    const errors: error[] = []
+    this.validateTimestampFormat(ctx, errors)
+    this.validateSignatureFormat(ctx, errors)
+    if (!errors.length) {
+      await this.validateSignatureAddress(ctx, errors)
+    }
+    if (errors.length) ctx.throw(BAD_REQUEST, {errors})
+  }
 
-  static validateForFindOne: publicValidateFunction = async (ctx) => {
-    await FilesValidator.validateFileExists(ctx)
+  static async validateForFindOne(ctx: Context) {
+    await this.validateFileExists(ctx)
   }
 
-  static validateForFindAll: publicValidateFunction = async (ctx) => {
+  static validateForFindAll(ctx: Context) {
     const errors: error[] = []
     const { owner, page, pageSize } = ctx.params
-    if (typeof owner !== 'undefined' && !isAddress(owner)) {
-      errors.push({ owner: 'Given address is not valid' })
+    if (typeof owner !== 'undefined' && !utils.isAddress(owner)) {
+      errors.push({ owner: 'Given owner address is not valid' })
     }
     if (typeof page !== 'undefined' && (typeof page != 'number' || page < 0)) {
       errors.push({ page: 'Given page is not valid' })
     }
     if (typeof pageSize !== 'undefined' && (typeof pageSize != 'number' || pageSize < 1)) {
       errors.push({ pageSize: 'Given pageSize is not valid' })
     }
+    if (errors.length) ctx.throw(BAD_REQUEST, {errors})
   }
 
-  static validateFileExists: publicValidateFunction = async (ctx) => {
+  static async validateFileExists(ctx: Context) {
     const { cid } = ctx.params
     if (!await FileMeta.exists({cid})) {
       ctx.throw(NOT_FOUND, { errors: [ { file: 'Not found' } ] })
     }
   }
+
+  private static validateTimestampFormat(ctx: Context, errors: error[]) {
+    const { timestamp } = ctx.request.body
+    if (!timestamp) {
+      errors.push({timestamp: 'A timestamp must be given'})
+    }
+    else if (Number(timestamp) > Date.now()) {
+      errors.push({timestamp: 'Given timestamp is not valid'})
+    }
+    else if (Number(timestamp) < Date.now()-SIGNATURE_TIMESTAMP_EXPIRES) {
```

Comment:
I thought this was a dash initially 😂 
```suggestion
    else if (Number(timestamp) < Date.now() - SIGNATURE_TIMESTAMP_EXPIRES) {
```